packages/firestore/test/unit/specs/listen_spec.test.ts

```diff
@@ -703,53 +695,103 @@ describeSpec('Listens:', [], () => {
     const query1 = Query.atPath(path('collection'));
     const query2 = Query.atPath(path('collection'));
 
-    return client(0)
-      .userListens(query1)
-      .watchAcksFull(query1, 1000)
-      .expectEvents(query1, {})
-      .client(1)
-      .userListens(query1)
-      .expectEvents(query1, {})
-      .client(0)
-      .disableNetwork()
-      .expectEvents(query1, { fromCache: true })
-      .client(1)
-      .expectEvents(query1, { fromCache: true })
-      .client(2)
-      .userListens(query1)
-      .expectEvents(query1, { fromCache: true })
-      .userListens(query2)
-      .expectEvents(query2, { fromCache: true });
+    return (
+      client(0)
+        .userListens(query1)
+        .watchAcksFull(query1, 1000)
+        .expectEvents(query1, {})
+        .client(1)
+        // Prevent client 0 from releasing its primary lease.
+        .disableNetwork()
+        .userListens(query1)
+        .expectEvents(query1, {})
+        .client(0)
+        .disableNetwork()
+        .expectEvents(query1, { fromCache: true })
+        .client(2)
+        .userListens(query1)
+        .expectEvents(query1, { fromCache: true })
+        .userListens(query2)
+        .expectEvents(query2, { fromCache: true })
+    );
   });
 
   specTest(
-    "Secondary client's online state is ignored",
+    'New client becomes primary if no client is online',
     ['multi-client'],
     () => {
       const query = Query.atPath(path('collection'));
-      const docA = doc('collection/a', 2000, { key: 'a' });
 
       return client(0)
-        .becomeVisible()
-        .client(1)
         .userListens(query)
-        .expectEvents(query, { fromCache: true })
-        .client(0)
-        .expectListen(query)
         .watchAcksFull(query, 1000)
+        .expectEvents(query, {})
         .client(1)
+        .userListens(query)
         .expectEvents(query, {})
-        .disableNetwork() // Ignored since this is the secondary client.
         .client(0)
-        .watchSends({ affects: [query] }, docA)
-        .watchSnapshots(2000)
+        .disableNetwork()
+        .expectEvents(query, { fromCache: true })
         .client(1)
-        .expectEvents(query, { added: [docA] })
+        .expectEvents(query, { fromCache: true })
+        .client(2)
+        .expectListen(query, 'resume-token-1000')
+        .expectPrimaryState(true)
+        .watchAcksFull(query, 2000)
         .client(0)
-        .disableNetwork()
+        .expectEvents(query, {})
         .client(1)
+        .expectEvents(query, {});
+    }
+  );
+
+  specTest(
+    "Secondary client's online state is ignored",
+    ['multi-client'],
+    () => {
+      const query = Query.atPath(path('collection'));
+      const docA = doc('collection/a', 2000, { key: 'a' });
+
+      return (
```

Comment:
paren

Reply:
This is `prettier` since the test case has an inline comment. Pretty, isn't it?